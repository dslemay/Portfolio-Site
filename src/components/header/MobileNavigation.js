import React, { Component } from 'react';
import PropTypes from 'prop-types';
import MenuIcon from './MenuIcon';
import NavigationLinks from './NavigationLinks';

class MobileNavigation extends Component {
  constructor() {
    super();

    this.state = {
      isOpen: false,
    };

    this.toggleOpen = this.toggleOpen.bind(this);
  }

  toggleOpen() {
    const { isOpen } = this.state;
    this.setState({ isOpen: !isOpen });
  }

  render() {
    return (
      <div style={{ position: 'relative', height: '65px' }}>
        <MenuIcon isOpen={this.state.isOpen} toggleOpen={this.toggleOpen} />
        <NavigationLinks mobile isOpen={this.state.isOpen} home={this.props.home} />
      </div>
    );
  }
}

MobileNavigation.propTypes = {
  home: PropTypes.bool.isRequired,
};

export default MobileNavigation;
