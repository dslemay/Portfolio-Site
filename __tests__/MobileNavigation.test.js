import React from 'react';
import { MemoryRouter } from 'react-router';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import MobileNavigation from '../src/components/header/MobileNavigation';

describe('MobileNavigation', () => {
  it('renders properly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <MobileNavigation />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('toggleOpen method toggles isOpen boolean state', () => {
    const wrapper = mount(<MobileNavigation />);
    wrapper.instance().toggleOpen();
    expect(wrapper.state().isOpen).toBe(true);
    wrapper.instance().toggleOpen();
    expect(wrapper.state().isOpen).toBe(false);
  });
});
