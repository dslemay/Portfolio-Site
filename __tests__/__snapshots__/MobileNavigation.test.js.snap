// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileNavigation renders properly 1`] = `
<div
  style={
    Object {
      "height": "65px",
      "position": "relative",
    }
  }
>
  <div
    className="css-5eaigi css-1e60agf0"
    onClick={[Function]}
  >
    <div
      className="css-m56tt4 css-1e60agf1"
    >
      <span
        className="css-190cys6 css-1e60agf2"
      />
      <span
        className="css-1nmu6z9 css-1e60agf3"
      />
      <span
        className="css-g78e9d css-1e60agf4"
      />
    </div>
  </div>
  <div
    className="css-1xxfl7e css-hn1q840"
  >
    <MockedLink
      className="css-hur3c3"
      to="/about"
    >
      About
    </MockedLink>
    <MockedLink
      className="css-hur3c3"
      to="/work"
    >
      Work
    </MockedLink>
    <MockedLink
      className="css-hur3c3"
      to="/contact"
    >
      Contact
    </MockedLink>
    <MockedLink
      className="css-hur3c3"
      to="/blog"
    >
      Blog
    </MockedLink>
  </div>
</div>
`;
